crypto/bn/bn_recp.c

```diff
@@ -63,6 +63,7 @@ void BN_RECP_CTX_init(BN_RECP_CTX *recp)
     bn_init(&(recp->N));
     bn_init(&(recp->Nr));
     recp->num_bits = 0;
+    recp->shift = 0;
```

Comment:
You're not very consistent here.  First you fix a bug because it's not initialized at all, and then you use zalloc, so you fixed it twice.  We might want to fix this in the stable branches too.